File: Java/src/main/java/QuestionBank/Solution1001_1100/Solution1051.java
package QuestionBank.Solution1001_1100;

import java.util.Arrays;

/**
 * @author windows
 */
public class Solution1051 {
    public int heightChecker(int[] heights) {
        int[] res = new int[heights.length];
        for (int i = 0; i < heights.length; i++) {
            res[i] = heights[i];
        }
        Arrays.sort(res);
        int count = 0;
        for (int i = 0; i < res.length; i++) {
            if (res[i] != heights[i]){
                count++;
            }
        }
        return count;
    }
    
    
    public static void main(String[] args) {
        Solution1051 solution1051 = new Solution1051();
    }
}

File: C++/src/Solution1001_1100/Solution1051.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1051.h"
#include <functional>
#include <algorithm>
#include <iostream>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;


int heightChecker(vector<int>& heights) {
  vector<int> res(heights);
  sort(heights.begin(), heights.end());
  int count = 0;
  for (int i = 0; i < heights.size(); ++i) {
    if (res[i] != heights[i]){
      count++;
    }
  }
  return count;
}
